src/utils/batched-updates.js

```diff
@@ -8,23 +8,33 @@ import {
 import defaults from '../defaults';
 import supports from './supported-features';
 
-let isInsideBatchedSchedule = false;
+let batchedScheduled = false;
+
+let batched = [];
+
+const executeBatched = () => {
+  unstable_batchedUpdates(() => {
+    while (batched.length) {
+      const currentBatched = batched;
+      batched = [];
+      currentBatched.forEach((fn) => fn());
```

Comment:
This now looks a lot like the queue logic in `schedule` https://github.com/atlassian/react-sweet-state/blob/master/src/utils/schedule.js ... Should we strip that and handle everything inside `batch`? 🤔

Reply:
Sounds logical, batch can handle queue, but there will be a different in behavior
- now `schedule` executed all notifications at once
- then `batch` will execute all notification and state updates in FIFO
- for react it might not matter, we are in `batchedUpdates`
- but for callbacks it might matter, as before notification can arrive _before_ state got updated, and now it will reach _after_
  - need to find a way to write a test indicating this behavior change, if it exists